trl/trainer/grpo_trainer.py

```diff
@@ -839,22 +846,34 @@ def _generate_and_score_completions(
         self._metrics[mode]["reward_std"].append(std_grouped_rewards.mean().item())
 
         if (
-            self.log_completions
-            and self.state.global_step % self.args.logging_steps == 0
-            and "wandb" in self.args.report_to
+            self.log_completions_steps is not None
+            and self.log_completions_steps > 0
+            and self.state.global_step % self.log_completions_steps == 0
         ):
-            import pandas as pd
-
-            # For logging
-            table = {
-                "step": [str(self.state.global_step)] * len(rewards),
-                "prompt": gather_object(prompts_text),
-                "completion": gather_object(completions_text),
-                "reward": rewards.tolist(),
-            }
-            df = pd.DataFrame(table)
-
-            if wandb.run is not None and self.accelerator.is_main_process:
+            prompts_to_log: list[str] = gather_object(prompts_text)
+            completions_to_log: list[str] = gather_object(completions_text)
+            pairs_to_log = random.sample(list(zip(prompts_to_log, completions_to_log)), 5)
```

Comment:
similarly, I'd go for full logging. We can still add a param to limit the number of lines to be logged later.